activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -529,12 +529,27 @@ def default_index_type?(index) # :nodoc:
         index.using == :btree || super
       end
 
-      def insert_fixtures(*)
-        without_sql_mode("NO_AUTO_VALUE_ON_ZERO") { super }
+      def insert_fixtures(fixture_set, tables_to_delete = [])
+        iterate_over_results = -> { while raw_connection.next_result; end; }
```

Comment:
Do no other adapters need something like this?

It doesn't feel like great layering for `insert_fixtures` to be touching `raw_connection` at all, really.

Reply:
No, MySQL is the only backend which forces you to care about what's in the network buffers, and manually flush it to avoid errors.

Reply:
Okay, cool. As I'm paranoid, I think I'd like to have a test that does a compound fixture load with an insert that succeeds followed by one that fails (inserting null into a not null column, say), and confirms we get the exception we're expecting. i.e., that all drivers [including out-of-tree] not only receive a compound result set without stuffing up the connection, but actually notice if a later result contains an error.

Reply:
For reference, the behavior that most other backends provide is that all queries are executed, but only the return value of the final query is provided. This is almost always handled at the client level, it's rare for the server to actually accept semicolon-delimited SQL strings (MySQL is the only one in this case AFAIK).

I agree we should have the test you've described though.

Reply:
Ack, will take care of writing the test. Thanks for the explanations

Reply:
Done in https://github.com/rails/rails/pull/31422/commits/7f8b345022887d1b7df3f486f9637a4aa7be1941